### PR TITLE
 Skip to prevote step if all pre-commits are collected

### PR DESCRIPTION
**Pull request by majecty**

### CODE REVIEW

File: core/src/consensus/tendermint/mod.rs [hunk 1]
```diff
@@ -435,6 +440,13 @@ impl Tendermint {
                 ctrace!(ENGINE, "Transition to {:?} triggered.", step);
                 self.to_step(step);
             }
+        } else if vote_step.step == Step::Precommit
+            && self.height() - 1 == vote_step.height
```
Comment (sgkim126) [inline core/src/consensus/tendermint/mod.rs]: Is it okay with the genesis block?
The hight of the genesis block is 0.

Comment (jhs7jhs) [reply core/src/consensus/tendermint/mod.rs]: `self.height()` has positive value always.

Comment (sgkim126) [reply core/src/consensus/tendermint/mod.rs]: You are right. it's always a positive value.
I checked that it's the height to vote, not the current height.

**Comment by sgkim126** [comment]
@majecty The change in the second commit is already applied to master.
https://github.com/CodeChain-io/codechain/commit/a9957f982b2ddf4f284eaad77546aacd4460d7c2#diff-1c1e4a7a7118105177e1800187ceca34R124

**Pull request merged**